packages/design-system/src/components/tree/tree-node.tsx

```diff
@@ -0,0 +1,306 @@
+
+  const handleAnimationEnd = useCallback(
+    (event: React.AnimationEvent<HTMLDivElement>) => {
+      if (event.target === collapsibleContentRef.current) {
+        onExpandTransitionEnd?.();
+      }
+    },
+    [onExpandTransitionEnd]
+  );
+
+  // If ther's no animation, we need to manually trigger onExpandTransitionEnd
+  const prevIsExpanded = useRef(isExpanded);
+  useEffect(() => {
+    if (animate === false && isExpanded !== prevIsExpanded.current) {
+      onExpandTransitionEnd?.();
+    }
+    prevIsExpanded.current = isExpanded;
+  }, [animate, onExpandTransitionEnd, isExpanded]);
+
+  return (
+    <Collapsible.Root
+      open={isExpanded}
+      onOpenChange={(isOpen) => setIsExpanded?.(itemData, isOpen)}
+      data-drop-target-id={itemData.id}
+    >
+      <ItemWrapper
+        isSelected={isSelected}
+        parentIsSelected={parentIsSelected}
+        enableHoverState={forceHoverStateAtItem === undefined}
+        forceHoverState={forceHoverStateAtItem === itemData.id}
+      >
+        {/* We want the main ItemButton to take the entire space,
+         * and then position the collapsible trigger on top of it using absolute positionning.
+         * When user clicks anywhere on a tree item, they should either hit the main button or the trigger.
+         */}
+
+        <ItemButton
+          type="button"
+          data-drag-item-id={itemData.id}
+          data-item-button-id={itemData.id}
+          onFocus={makeSelected}
+        >
+          <NestingLines isSelected={isSelected} level={level} />
+          {isAlwaysExpanded === false && <TriggerPlaceholder />}
+          {renderItem({ data: itemData, isSelected })}
+        </ItemButton>
+
+        {shouldRenderExpandButton && (
+          <CollapsibleTrigger
+            style={{ left: (level - 1) * INDENT + ITEM_PADDING }}
+            // We don't want a separate focusable control inside a tree item.
+            // tabIndex makes it skipped over when tabbing.
+            // It still can be focused using mouse, but we handle this elsewhere.
+            tabIndex={-1}
+          >
+            {isExpanded ? <TriangleDownIcon /> : <TriangleRightIcon />}
+          </CollapsibleTrigger>
+        )}
+      </ItemWrapper>
+      {isExpandable && (
+        <CollapsibleContent
+          onAnimationEnd={handleAnimationEnd}
+          ref={collapsibleContentRef}
+        >
+          {
+            // CollapsibleContent doesn't render children when collapsed.
+            // No need to worry about optimizing this.
+            itemChildren.map((child) => (
+              <TreeNode
+                key={child.id}
+                itemData={child}
+                level={level + 1}
+                parentIsSelected={isSelected || parentIsSelected}
+                {...commonProps}
+              />
+            ))
+          }
+        </CollapsibleContent>
+      )}
+    </Collapsible.Root>
+  );
+};
+
+const Label = styled(__DEPRECATED__Text, {
```

Comment:
lets use the new Text component right away?

Reply:
I use `variant={isSelected ? "loContrast" : "contrast"}` from the old Text, and the new one doesn't support this.

I'd kept `__DEPRECATED__Text` here for now and replace it with `Text` when it will be replaced everywhere in the codebase.

Reply:
lets see what we need to add to support it, its easy,  we should not use anything that has deprecated in it as a rule

Reply:
I think this would work, adding to variants: 

```ts
```